### PR TITLE
feat: non-permissioned networks, when a dapp finds itself on a global network for which it doesn't have a granted network permission

### PR DESCRIPTION
**Pull request by EtherWizard33**
non-permissioned networks, when a dapp finds itself on a global network for which it doesn't have a granted network permission 

## **Description**

Figma: https://www.figma.com/design/njDnVDROVuIwLbjUjkUYuO/MC-Mobile-Amon-Hen?node-id=3109-157108&node-type=canvas&m=dev
## **Related issues**

Contributes to: https://github.com/MetaMask/MetaMask-planning/issues/3598

## **Manual testing steps**

1. Start with a global network and a dapp connected to it
2. From the home screen, switch the network
3. Go back to the dapp, the bottom sheet will come up. 

This works also with multiple dapps open at once, the bottom sheet comes up the next time the dapp is in focus.

## **Screenshots/Recordings**

| Before  (switch to non permitted change was not handled)    | After  (handled)  |
|--------------|--------------|
| Screenshot 2024-04-18 at 3 56 43 PM |Screenshot 2024-04-18s at 3 56 43 PM |

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by vinnyhoward** [comment]
LGTM!✅ Very clean

**Review by vinnyhoward — DISMISSED** [review]
LGTM!✅ Very clean

**Review by vinnyhoward — DISMISSED** [review]

**Review by vinnyhoward — DISMISSED** [review]

**Review by tommasini — DISMISSED** [review]
I believe if we have the time the comments that I left are important to be addressed, broken my perspective it's okay to merge it and create an issue with those to address after the feature is shipped 
LGTM

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -1533,6 +1540,68 @@ export const BrowserTab = (props) => {
     [props.linkType],
   );
 
+  const checkTabPermissions = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (EtherWizard33) [reply app/components/Views/BrowserTab/index.js:1543]: yes, we are aligned on this, I will refer to [my other comment above for details](https://github.com/MetaMask/metamask-mobile/pull/12212#discussion_r1852046413)

**Review by tommasini — APPROVED** [review]

**Review by vinnyhoward — APPROVED** [review]

**Pull request merged**